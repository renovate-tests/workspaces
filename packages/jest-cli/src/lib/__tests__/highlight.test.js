/**
* Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
*
* This source code is licensed under the BSD-style license found in the
* LICENSE file in the root directory of this source tree. An additional grant
* of patent rights can be found in the PATENTS file in the same directory.
*
* @flow
*/

'use strict';

const highlight = require('../highlight');

const rootDir = '/Users/foo/dev/jest';
const rawPath = rootDir + '/jest-cli/__tests__/watch.test.js';
const relativePaths = [
  'jest-cli/__tests__/watch.test.js',
  '...t-cli/__tests__/watch.test.js',
  '.../__tests__/watch.test.js',
  '...sts__/watch.test.js',
  '...watch.test.js',
  '....test.js',
  '....js',
  './watch.test.js',
];

it('highlight the trimmed part when there is only a rootDir match', () => {
  relativePaths
    .map(trimmed => highlight(rawPath, trimmed, 'Users/foo', rootDir))
    .forEach(output => expect(output).toMatchSnapshot());
});

it('highlights the trimmed part there is a non visible match', () => {
  relativePaths
    .filter(filePath => filePath.startsWith('...'))
    .map(trimmed => highlight(rawPath, trimmed, 'jest-cli', rootDir))
    .forEach(output => expect(output).toMatchSnapshot());
});

it('dims everything when there is no match', () => {
  relativePaths
    .map(trimmed => highlight(rawPath, trimmed, 'nomatch', rootDir))
    .forEach(output => expect(output).toMatchSnapshot());
});

it('highlights everything when there is a full match', () => {
  relativePaths
    .map(trimmed => highlight(rawPath, trimmed, 'User.*js', rootDir))
    .forEach(output => expect(output).toMatchSnapshot());
});

it('highlights part of file name when there is a partially match of the file name', () => {
  relativePaths
    .map(trimmed => highlight(rawPath, trimmed, 'watch', rootDir))
    .forEach(output => expect(output).toMatchSnapshot());
});
