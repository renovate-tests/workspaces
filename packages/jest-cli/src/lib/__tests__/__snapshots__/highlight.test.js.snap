// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dims everything when there is no match 1`] = `"<dim>jest-cli/__tests__/watch.test.js"`;

exports[`dims everything when there is no match 2`] = `"<dim>...t-cli/__tests__/watch.test.js"`;

exports[`dims everything when there is no match 3`] = `"<dim>.../__tests__/watch.test.js"`;

exports[`dims everything when there is no match 4`] = `"<dim>...sts__/watch.test.js"`;

exports[`dims everything when there is no match 5`] = `"<dim>...watch.test.js"`;

exports[`dims everything when there is no match 6`] = `"<dim>....test.js"`;

exports[`dims everything when there is no match 7`] = `"<dim>....js"`;

exports[`dims everything when there is no match 8`] = `"<dim>./watch.test.js"`;

exports[`highlight the trimmed part when there is only a rootDir match 1`] = `"<dim>jest-cli/__tests__/watch.test.js"`;

exports[`highlight the trimmed part when there is only a rootDir match 2`] = `"</>...</><dim>t-cli/__tests__/watch.test.js"`;

exports[`highlight the trimmed part when there is only a rootDir match 3`] = `"</>...</><dim>/__tests__/watch.test.js"`;

exports[`highlight the trimmed part when there is only a rootDir match 4`] = `"</>...</><dim>sts__/watch.test.js"`;

exports[`highlight the trimmed part when there is only a rootDir match 5`] = `"</>...</><dim>watch.test.js"`;

exports[`highlight the trimmed part when there is only a rootDir match 6`] = `"</>...</><dim>.test.js"`;

exports[`highlight the trimmed part when there is only a rootDir match 7`] = `"</>...</><dim>.js"`;

exports[`highlight the trimmed part when there is only a rootDir match 8`] = `"</>./</><dim>watch.test.js"`;

exports[`highlights everything when there is a full match 1`] = `"</>jest-cli/__tests__/watch.test.js</>"`;

exports[`highlights everything when there is a full match 2`] = `"</>...t-cli/__tests__/watch.test.js</>"`;

exports[`highlights everything when there is a full match 3`] = `"</>.../__tests__/watch.test.js</>"`;

exports[`highlights everything when there is a full match 4`] = `"</>...sts__/watch.test.js</>"`;

exports[`highlights everything when there is a full match 5`] = `"</>...watch.test.js</>"`;

exports[`highlights everything when there is a full match 6`] = `"</>....test.js</>"`;

exports[`highlights everything when there is a full match 7`] = `"</>....js</>"`;

exports[`highlights everything when there is a full match 8`] = `"</>./watch.test.js</>"`;

exports[`highlights part of file name when there is a partially match of the file name 1`] = `"<dim>jest-cli/__tests__/</>watch</><dim>.test.js"`;

exports[`highlights part of file name when there is a partially match of the file name 2`] = `"<dim>...t-cli/__tests__/</>watch</><dim>.test.js"`;

exports[`highlights part of file name when there is a partially match of the file name 3`] = `"<dim>.../__tests__/</>watch</><dim>.test.js"`;

exports[`highlights part of file name when there is a partially match of the file name 4`] = `"<dim>...sts__/</>watch</><dim>.test.js"`;

exports[`highlights part of file name when there is a partially match of the file name 5`] = `"<dim>...</>watch</><dim>.test.js"`;

exports[`highlights part of file name when there is a partially match of the file name 6`] = `"</>...</><dim>.test.js"`;

exports[`highlights part of file name when there is a partially match of the file name 7`] = `"</>...</><dim>.js"`;

exports[`highlights part of file name when there is a partially match of the file name 8`] = `"<dim>./</>watch</><dim>.test.js"`;

exports[`highlights the trimmed part there is a non visible match 1`] = `"</>...t-cli</><dim>/__tests__/watch.test.js"`;

exports[`highlights the trimmed part there is a non visible match 2`] = `"</>...</><dim>/__tests__/watch.test.js"`;

exports[`highlights the trimmed part there is a non visible match 3`] = `"</>...</><dim>sts__/watch.test.js"`;

exports[`highlights the trimmed part there is a non visible match 4`] = `"</>...</><dim>watch.test.js"`;

exports[`highlights the trimmed part there is a non visible match 5`] = `"</>...</><dim>.test.js"`;

exports[`highlights the trimmed part there is a non visible match 6`] = `"</>...</><dim>.js"`;
